src/Sylius/Bundle/ResourceBundle/Controller/ParametersParser.php

```diff
@@ -81,6 +82,31 @@ private function parseRequestValue($parameter, Request $request)
         return $parameter;
     }
 
+    /**
+     * @param string $parameter
+     * @param Request $request
+     *
+     * @return mixed
+     */
+    private function getRequestValue($parameter, Request $request)
+    {
+        preg_match('/[a-z0-9]+/i', $parameter, $match);
```

Comment:
`/^(?P<variable>[^\[]++)(?P<property>(?:\[[^\]\[]++\])++)?$/` regexp might do better here, you will get both of the things you need using one `preg_match` call.

Reply:
You can access them through `$match['variable']` and `$match['property']`.